pace-util/pace/util/communicator.py

```diff
@@ -645,3 +518,211 @@ def get_vector_halo_updater(
             self._get_halo_tag(),
             self.timer,
         )
+
+    def _get_halo_tag(self) -> int:
+        self._last_halo_tag += 1
+        return self._last_halo_tag
+
+    @property
+    def boundaries(self) -> Mapping[int, Boundary]:
+        """boundaries of this tile with neighboring tiles"""
+        if self._boundaries is None:
+            self._boundaries = {}
+            for boundary_type in constants.BOUNDARY_TYPES:
+                boundary = self.partitioner.boundary(boundary_type, self.rank)
+                if boundary is not None:
+                    self._boundaries[boundary_type] = boundary
+        return self._boundaries
+
+
+class TileCommunicator(Communicator):
+    """Performs communications within a single tile or region of a tile"""
+
+    def __init__(
+        self,
+        comm,
+        partitioner: TilePartitioner,
+        force_cpu: bool = False,
+        timer: Optional[Timer] = None,
+    ):
+        """Initialize a TileCommunicator.
+
+        Args:
+            comm: communication object behaving like mpi4py.Comm
+            partitioner: tile partitioner
+            force_cpu: force all communication to go through central memory
+            timer: Time communication operations.
+        """
+        super(TileCommunicator, self).__init__(
+            comm, partitioner, force_cpu=force_cpu, timer=timer
+        )
+        self.partitioner: TilePartitioner = partitioner
+
+    @property
+    def tile(self):
+        return self
+
+    def start_halo_update(
+        self, quantity: Union[Quantity, List[Quantity]], n_points: int
+    ) -> HaloUpdater:
+        """Start an asynchronous halo update on a quantity.
+
+        Args:
+            quantity: the quantity to be updated
+            n_points: how many halo points to update, starting from the interior
+
+        Returns:
+            request: an asynchronous request object with a .wait() method
+        """
+        if self.partitioner.layout[0] < 3 or self.partitioner.layout[1] < 3:
+            raise NotImplementedError()
```

Comment:
Maybe a comment here to explain why it's not implemented

Reply:
(and below)

Reply:
Thanks, updated!